webapp/channels/src/store/index.ts

```diff
@@ -14,15 +16,34 @@ import {clearUserCookie} from 'actions/views/cookie';
 import appReducers from 'reducers';
 import {getBasePath} from 'selectors/general';
 
+import type {GlobalState} from 'types/store';
+
 function getAppReducers() {
     return require('../reducers'); // eslint-disable-line global-require
 }
 
+declare global {
+    interface Window {
+        Observable: typeof Observable;
+    }
+}
+
+type DeepPartial<T> = T extends object ? {
```

Comment:
We have a more robust version of this type in `@mattermost/types/utilities`. It also supports arrays and some other types like `Set`